src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,134 @@
+    void register_nullLogin_Exception_notOk() {
+        RegistrationException invalidDataException = assertThrows(RegistrationException.class,
+                () -> {
+                regService.register(new User(NULL_LOGIN, VALID_PWD, VALID_AGE));
+                });
+        assertEquals("Login can't be null", invalidDataException.getMessage());
+    }
+
+    @Test
+    void register_nullAge_Exception_notOk() {
+        RegistrationException invalidDataException = assertThrows(RegistrationException.class,
+                () -> {
+                regService.register(new User(VALID_LOGIN1, VALID_PWD, NULL_AGE));
+                });
+        assertEquals("Age can't be null", invalidDataException.getMessage());
+    }
+
+    @Test
+    void register_invalidLogin_Exception_notOk() {
+        RegistrationException invalidDataException = assertThrows(RegistrationException.class,
+                () -> {
+                regService.register(INVALID_LOGIN_USER);
+                });
+        assertEquals("Not valid login length: "
+                + INVALID_LOGIN_USER.getLogin().length()
+                + ". Min allowed login length is " + MIN_LENGTH, invalidDataException.getMessage());
+    }
+
+    @Test
+    void register_existUser_Exception_notOk() {
+        Storage.people.add(VALID_USER);
+        RegistrationException invalidDataException = assertThrows(RegistrationException.class,
+                () -> {
+                    regService.register(VALID_USER);
+                });
+        assertEquals("User with this login already exists", invalidDataException.getMessage());
+    }
+
+    @Test
+    void register_invalidPassword_Exception_notOk() {
+        RegistrationException invalidDataException = assertThrows(RegistrationException.class,
+                () -> {
+                regService.register(INVALID_PWD_USER);
+                });
+        assertEquals("Not valid password length: "
+                + INVALID_PWD_USER.getPassword().length()
+                + ". Min allowed password length is "
+                + MIN_LENGTH, invalidDataException.getMessage());
+    }
+
+    @Test
+    void register_invalidAge_Exception_notOk() {
+        RegistrationException invalidDataException = assertThrows(RegistrationException.class,
+                () -> {
+                regService.register(INVALID_AGE_USER);
+                });
+        assertEquals("Not valid age: "
+                + INVALID_AGE_USER.getAge()
+                + ". Min allowed age is " + MIN_AGE, invalidDataException.getMessage());
+    }
+
+    @Test
+    void register_oneUser_Ok() {
+        User validUser = regService.register(VALID_USER);
+        assertEquals(validUser, VALID_USER);
+    }
+
+    @Test
+    void register_twoUsers_Ok() {
+        User validUser2 = regService.register(VALID_USER2);
+        User validUser3 = regService.register(VALID_USER3);
+        assertEquals(validUser2, VALID_USER2);
+        assertEquals(validUser3, VALID_USER3);
+    }
+
```

Comment:
it seems to me that there should be another method that is executed at the end of all tests and cleans the storage.